discord_bots/cogs/in_progress_game.py

```diff
@@ -111,151 +139,281 @@ def get_player_and_in_progress_game(
-    ) -> bool:
+    async def cancelgame_callback(self, interaction: Interaction, game_id: str) -> bool:
         if not interaction.response.is_done():
             await interaction.response.defer(ephemeral=True)
         if _lock.locked():
             await interaction.followup.send(
-                embed=discord.Embed(
-                    description="⏳ A **Finish** is already in progress, please wait...",
-                    color=discord.Color.yellow(),
+                embed=Embed(
+                    description="⏳ A **Cancel** is already in progress, please wait...",
+                    color=Colour.yellow(),
                 ),
                 ephemeral=True,
             )
         async with _lock:
-            session: sqlalchemy.orm.Session
+            session: SQLAlchemySession
             with Session() as session:
-                result = self.get_player_and_in_progress_game(
-                    session, interaction.user.id, game_id
+                game = (
+                    session.query(InProgressGame)
+                    .filter(InProgressGame.id.startswith(game_id))
+                    .first()
                 )
-                if result is None:
-                    embed = discord.Embed(
-                        color=discord.Colour.red(),
-                    )
-                    embed.description = "❌ You are not in this game"
+                if not game:
                     await interaction.followup.send(
-                        embed=embed,
+                        embed=Embed(
+                            description=f"❌ Game {short_uuid(game_id)} does not exist",
+                            colour=Colour.red(),
+                        ),
                         ephemeral=True,
                     )
                     return False
-
-                game_player: InProgressGamePlayer = result[0]
-                game: InProgressGame = result[1]
                 confirmation_buttons = ConfirmationView(interaction.user.id)
                 confirmation_buttons.message = await interaction.followup.send(
-                    embed=discord.Embed(
-                        description=f"⚠️ Are you sure you want to **Finish** game **{short_uuid(game.id)}** as a **{outcome}**?⚠️",
-                        color=discord.Colour.yellow(),
+                    embed=Embed(
+                        description=f"⚠️ Are you sure you want to **Cancel** game **{short_uuid(game.id)}**?⚠️",
+                        color=Colour.yellow(),
                     ),
                     view=confirmation_buttons,
                     ephemeral=True,
                 )
                 await confirmation_buttons.wait()
                 if not confirmation_buttons.value:
                     return False
-                is_finished = await self.finish_in_progress_game(
-                    session, interaction, outcome, game_player, game
+                is_game_finished = await self.cancel_in_progress_game(
```

Comment:
Good catch, since this can be confused with the internal view.is_finished field used by discord.py